lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1916,22 +1916,36 @@ void TypeChecker::checkIgnoredExpr(Expr *E) {
       .highlight(E->getSourceRange());
     return;
   }
-    
-  // Always complain about 'try?'.
-  if (auto *OTE = dyn_cast<OptionalTryExpr>(valueE)) {
-    DE.diagnose(OTE->getTryLoc(), diag::expression_unused_optional_try)
-      .highlight(E->getSourceRange());
-    return;
-  }
 
   if (auto *LE = dyn_cast<LiteralExpr>(valueE)) {
     diagnoseIgnoredLiteral(Context, LE);
     return;
   }
+  
+  ApplyExpr *call = nullptr;
+  
+  // Unwrap optional try to get inner function calls.
+  {
+    auto nextExpr = valueE;
+    while(true) {
+      if (nextExpr == nullptr) {
```

Comment:
wonder if it would work the same if you move this check into the `while` condition? i.e:
```cpp
while(nextExpr != nullptr)
```

Reply:
@Jamezzzb Fixed. Thanks